tools/lib/gzipTileset.js

```diff
@@ -0,0 +1,110 @@
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        var gzip = defaultValue(options.gzip, true);
+        var tilesOnly = defaultValue(options.tilesOnly, false);
+        var verbose = defaultValue(options.verbose, false);
+
+        if (!defined(inputDirectory)) {
+            reject(new DeveloperError('inputDirectory is required'));
+        }
+        inputDirectory = path.normalize(inputDirectory);
+        outputDirectory = path.normalize(defaultValue(outputDirectory,
+            path.join(path.dirname(inputDirectory), path.basename(inputDirectory) + '-' + (gzip ? 'gzipped' : 'gunzipped'))));
+
+        if (verbose) {
+            console.log('Input directory: ' + inputDirectory);
+            console.log('Output directory: ' + outputDirectory);
+        }
+
+        var files = [];
+        fsExtra.walk(inputDirectory)
+            .on('data', function (item) {
+                if (!item.stats.isDirectory()) {
+                    files.push(path.relative(inputDirectory, item.path));
+                }
+            })
+            .on('end', function () {
+                if (verbose) {
+                    console.log(files.length + ' files found.');
+                    console.log((gzip ? 'Compressing' : 'Uncompressing') + ' files...');
+                }
+                Promise.map(files, function (file) {
+                    var inputFile = path.join(inputDirectory, file);
+                    var outputFile = path.join(outputDirectory, file);
+
+                    if (gzip && tilesOnly && !isTile(inputFile)) {
+                        return fsExtraCopy(inputFile, outputFile);
+                    }
+
+                    return isGzipped(inputFile)
+                        .then(function(fileIsGzipped) {
+                            if (fileIsGzipped === gzip) {
+                                return fsExtraCopy(inputFile, outputFile);
+                            }
+                            return fsExtraEnsureDir(path.dirname(outputFile))
+                                .then(function () {
+                                    var inp = fsExtra.createReadStream(inputFile);
+                                    var out = fsExtra.createWriteStream(outputFile);
+                                    var operation = gzip ? zlib.createGzip() : zlib.createGunzip();
+                                    return streamToPromise(inp.pipe(operation).pipe(out));
+                                });
+
+                        });
+                }, {concurrency: 1024})
+                    .then(resolve)
+                    .catch(reject);
+            })
+            .on('error', reject);
+    });
+}
+
+function isTile(file) {
+    var extension = path.extname(file);
+    return extension === '.b3dm' ||
+           extension === '.i3dm' ||
+           extension === '.pnts' ||
+           extension === '.cmpt' ||
+           extension === '.vctr';
+}
+
+function isGzipped(file) {
```

Comment:
This could be in its own `@private` file instead of duplicating it in the tests.